### PR TITLE
feat: rule test supports function

### LINKED ISSUE #171: Rule test cannot be a function?
Like this:

```js
 module: {
    rules: [
      {
        test: () => /* some logic */,
...
```

I find that test function will be convert to an object? That's weird.

### PR DESCRIPTION
**Pull request by tinymins**
Fix #171.

### CODE REVIEW

File: test/Rule.js [hunk 1]
```diff
@@ -131,6 +131,15 @@ test('toConfig with values', t => {
   });
 });
 
+test('toConfig with test function', t => {
+  const rule = new Rule();
+  const test = s => s.indexOf('.js') >= 0;
```
Comment (eliperelman) [inline test/Rule.js]: Can you make this:

```js
const test = s => s.includes('.js');
```

File: src/Rule.js [hunk 2]
```diff
@@ -87,7 +87,11 @@ const Rule = Orderable(
       }
 
       if (!omit.includes('test') && 'test' in obj) {
-        this.test(obj.test instanceof RegExp ? obj.test : new RegExp(obj.test));
+        this.test(
+          obj.test instanceof RegExp || obj.test instanceof Function
```
Comment (eliperelman) [inline src/Rule.js]: Can you change `obj.test instanceof Function` to `typeof obj.test === 'function'`?

**Comment by tinymins** [comment]
@eliperelman Thanks for your advice, all changes applied.

**Comment by edmorley** [comment]
Published in [v6.1.0](https://www.npmjs.com/package/webpack-chain/v/6.1.0) :-)

**Pull request merged**